ros/src/inertial_sense_ros.cpp

```diff
@@ -535,21 +535,36 @@ bool InertialSenseROS::connect(float timeout)
 {
     uint32_t end_time = ros::Time::now().toSec() + timeout;
     auto ports_iterator = ports_.begin();
+    auto serialNum = 0;
 
     do {
         std::string cur_port = *ports_iterator;
         /// Connect to the IMX
         ROS_INFO("InertialSenseROS: Connecting to serial port \"%s\", at %d baud", cur_port.c_str(), baudrate_);
         sdk_connected_ = IS_.Open(cur_port.c_str(), baudrate_);
-        if (!sdk_connected_) {
+        if (sdk_connected_)
+        {
+            serialNum = IS_.DeviceInfo().serialNumber;
+        }
+        
+        if (!sdk_connected_) 
+        {
             ROS_ERROR("InertialSenseROS: Unable to open serial port \"%s\", at %d baud", cur_port.c_str(), baudrate_);
-            sleep(1); // is this a good idea?
-        } else {
+            usleep(500000); // is this a good idea?
```

Comment:
Is it a good idea?  You're sleeping for... 0.5ms?  It looks like it is scanning through available serial ports, so probably not the worst.  Ideally though, you wouldn't wait, you'd just jump directly to the next port.

It feels like there maybe a better way to write this, where you can iterate through each port, and on failure call `continue;` to jump directly to the next port.   Probably makes the most sense to do a `while()` for the timeout, and then to a `for (auto cur_port : ports_) { ... }` to iterate through all the ports.

This will give the benefit of ensuring that all ports get tried at least once, before it gives up.  As its written, it could timeout before it checks all the ports in the `ports_` list.

Reply:
I'd just like to say that the comment was made by Walt when he wrote this last year. 😄  I just halved the sleep time since it took a while. Unless it's really necessary I don't want to chop of this code right now. 🤷